inference-engine/tests/functional/plugin/cpu/single_layer_tests/scatter_ND_update.cpp

```diff
@@ -0,0 +1,173 @@
+                do {
+                    result << CommonTestUtils::vec2str(*itr);
+                } while (++itr != shape.second.end() && result << "_");
+            }
+            result << ")_";
+        }
+        result << "indices_values=" << CommonTestUtils::vec2str(indicesValues) << "_idx_precision=" << idxPrecision;
+        return result.str();
+    }
+
+protected:
+    void generate_inputs(const std::vector<ov::Shape>& targetInputStaticShapes) override {
+        inputs.clear();
+        const auto& funcInputs = function->inputs();
+        for (int i = 0; i < funcInputs.size(); ++i) {
+            const auto& funcInput = funcInputs[i];
+            const auto& inputPrecision = funcInput.get_element_type();
+            const auto& targetShape = targetInputStaticShapes[i];
+            ov::runtime::Tensor tensor;
+            if (i == 1) {
+                tensor = ov::runtime::Tensor{ inputPrecision, targetShape };
+                const auto indicesVals = std::get<0>(this->GetParam()).indicesValues;
+                if (inputPrecision == ElementType::i32) {
+                    auto data = tensor.data<std::int32_t>();
+                    for (size_t i = 0; i < tensor.get_size(); ++i) {
+                        data[i] = static_cast<std::int32_t>(indicesVals[i]);
+                    }
+                } else if (inputPrecision == ElementType::i64) {
+                    auto data = tensor.data<std::int64_t>();
+                    for (size_t i = 0; i < tensor.get_size(); ++i) {
+                        data[i] = indicesVals[i];
+                    }
+                } else {
+                    IE_THROW() << "GatherNDUpdate. Unsupported indices precision: " << inputPrecision;
+                }
+            } else {
+                if (inputPrecision.is_real()) {
+                    tensor = ov::test::utils::create_and_fill_tensor(inputPrecision, targetShape, 10, 0, 1000);
+                } else {
+                    tensor = ov::test::utils::create_and_fill_tensor(inputPrecision, targetShape);
+                }
+            }
+            inputs.insert({ funcInput.get_node_shared_ptr(), tensor });
+        }
+    }
+
+    void SetUp() override {
+        targetDevice = CommonTestUtils::DEVICE_CPU;
+        ScatterNDUpdateLayerParams scatterParams;
+        ElementType inputPrecision;
+        ElementType idxPrecision;
+        std::tie(scatterParams, inputPrecision, idxPrecision) = this->GetParam();
+        const auto inputShapes = scatterParams.inputShapes;
+        const auto indicesValues = scatterParams.indicesValues;
+
+        init_input_shapes(inputShapes);
+
+        auto dataParams = ngraph::builder::makeDynamicParams(inputPrecision, { inputDynamicShapes[0], inputDynamicShapes[2] });
+        auto indicesParam = ngraph::builder::makeDynamicParams(idxPrecision, { inputDynamicShapes[1] });
+        auto scatter = std::make_shared<ngraph::opset4::ScatterNDUpdate>(dataParams[0], indicesParam[0], dataParams[1]);
```

Comment:
please `set_friendly_name` for all params to be sure that they sorted in lexicographical order
same for other tests

Reply:
Done, thanks